host_applications/linux/apps/raspicam/imx219_modes.h

```diff
@@ -74,11 +74,11 @@ struct sensor_regs imx219_8MPix[] =
       {0x0170, 0x01},      // X_ODD_INC [2:0]
       {0x0171, 0x01},      // Y_ODD_INC [2:0]
       {0x0172, 0x03},
-      {0x0174, 0x00},      // Binning Mode H_A
-      {0x0175, 0x00},      // Binning Mode V_A
+      {0x0174, 0x01},      // Binning Mode H_A
+      {0x0175, 0x01},      // Binning Mode V_A
```

Comment:
Binning mode at 0x01 would be for the binned modes, not the 8MPix mode.
Look at line 126 of your gist. I think you've missed that raspistill will normally be running in the binned mode during preview and switches to 8MPix on capture. Easiest approach to avoid confusion is to add "-md 2" to force it to always use 8MPix mode.